src/project.test.ts

```diff
@@ -120,4 +130,147 @@ describe('project', () => {
       });
     });
   });
+  describe('resetUnreleasedPackagesChangelog', () => {
+    it('should reset changelog for packages with changes not included in release', async () => {
+      const project = buildMockProject({
+        rootPackage: buildMockPackage('monorepo'),
+        workspacePackages: {
+          a: buildMockPackage('a', {
+            hasChangesSinceLatestRelease: true,
+          }),
+          b: buildMockPackage('b', {
+            hasChangesSinceLatestRelease: false,
+          }),
+          c: buildMockPackage('c', {
+            hasChangesSinceLatestRelease: true,
+          }),
+        },
+      });
+
+      const restoreFilesSpy = jest.spyOn(repoModule, 'restoreFiles');
+
+      await restoreUnreleasedPackagesChangelog({
+        project,
+        defaultBranch: 'main',
+        releaseSpecification: {
+          packages: {
+            a: IncrementableVersionParts.minor,
+          },
+          path: '/path/to/release/specs',
+        },
+      });
+
+      expect(restoreFilesSpy).toHaveBeenCalledWith(
+        project.directoryPath,
+        'main',
+        ['/path/to/packages/c/CHANGELOG.md'],
+      );
+    });
```

Comment:
Should we also add a test for when a package is listed, but as "intentionally-skip"?

Reply:
@mcmire same as comment above intentionally-skip is not relevant here as this the final release specification which includes only packages to release